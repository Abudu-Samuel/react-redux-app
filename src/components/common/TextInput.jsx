/* eslint-disable jsx-a11y/label-has-for */
import React from 'react';
import PropTypes from 'prop-types';

const TextInput = ({
  name, label, onChange, placeholder, value, error, defaultValue
}) => (
  <div className="form-group">
    <label htmlFor={name}>{label}</label>
    <div className="field">
      <input
          type="text"
          name={name}
          className="form-control"
          placeholder={placeholder}
          defaultValue={defaultValue}
          onChange={onChange} />
      {error && <div className="alert alert-danger">{error}</div>}
    </div>
  </div>
);


TextInput.propTypes = {
  name: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  onChange: PropTypes.func,
  placeholder: PropTypes.string.isRequired,
  error: PropTypes.string
};

export default TextInput;