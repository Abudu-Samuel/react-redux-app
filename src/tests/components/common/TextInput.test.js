/* eslint-disable */
import TextInput from '../../../components/common/TextInput';

const config = (error) => {
  const props = {
    name: '',
    label: '',
    onChange: jest.fn(),
    placeholder: '',
    error,
    defaultValue: ''
  };
  return mount(<TextInput {...props} />);
};

let mountWrap;

describe('<TextInput />', () => {
  it('should render TextInput component', () => {
    mountWrap = config('');
    expect(toJson(mountWrap)).toMatchSnapshot();
  });

  it('should render error tag when an error occurs', () => {
    mountWrap = config('this field is required');
    expect(mountWrap.find('.alert-danger').length).toEqual(1);
    expect(mountWrap.find('.alert-danger').get(0).props.children).toEqual('this field is required');
  });
});