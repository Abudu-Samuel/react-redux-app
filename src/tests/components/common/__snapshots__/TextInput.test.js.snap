// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> should render TextInput component 1`] = `
<TextInput
  defaultValue=""
  error=""
  label=""
  name=""
  onChange={[MockFunction]}
  placeholder=""
>
  <div
    className="form-group"
  >
    <label
      htmlFor=""
    />
    <div
      className="field"
    >
      <input
        className="form-control"
        defaultValue=""
        name=""
        onChange={[MockFunction]}
        placeholder=""
        type="text"
      />
    </div>
  </div>
</TextInput>
`;
